src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzer.cs

```diff
@@ -326,16 +326,34 @@ void VerifySymbol(Action<Diagnostic> reportDiagnostic, ISymbol symbol, SyntaxNod
             {
                 RoslynDebug.Assert(entryBySymbol != null);
 
-                symbol = symbol.OriginalDefinition;
+                foreach (var currentSymbol in GetSymbolAndOverridenSymbols(symbol))
+                {
+                    if (entryBySymbol.TryGetValue(currentSymbol, out var entry))
+                    {
+                        reportDiagnostic(
+                            Diagnostic.Create(
+                                SymbolIsBannedAnalyzer.SymbolIsBannedRule,
+                                syntaxNode.GetLocation(),
+                                currentSymbol.ToDisplayString(SymbolDisplayFormat),
+                                string.IsNullOrWhiteSpace(entry.Message) ? "" : ": " + entry.Message));
+                        return;
+                    }
+                }
 
-                if (entryBySymbol.TryGetValue(symbol, out var entry))
+                static IEnumerable<ISymbol> GetSymbolAndOverridenSymbols(ISymbol symbol)
                 {
-                    reportDiagnostic(
-                        Diagnostic.Create(
-                            SymbolIsBannedAnalyzer.SymbolIsBannedRule,
-                            syntaxNode.GetLocation(),
-                            symbol.ToDisplayString(SymbolDisplayFormat),
-                            string.IsNullOrWhiteSpace(entry.Message) ? "" : ": " + entry.Message));
+                    ISymbol? currentSymbol = symbol.OriginalDefinition;
+
+                    while (currentSymbol != null)
+                    {
+                        yield return currentSymbol;
+
+                        // It's possible to have `IsOverride` true and yet have `GetOverriddeMember` returning false when the code is invalid
+                        // (e.g. base symbol is not marked as `virtual` or `abstract` and current symbol has the `overrides` modifier).
```

Comment:
Nice you handled this case, I was primarily looking for this case being covered in this review :-)